src/Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/NeighborPackagedData.hpp

```diff
@@ -0,0 +1,231 @@
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/BoundaryCorrections/Factory.hpp"
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/FiniteDifference/Reconstructor.hpp"
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/FiniteDifference/Tag.hpp"
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/Subcell/ComputeFluxes.hpp"
+#include "Evolution/Systems/GrMhd/ValenciaDivClean/System.hpp"
+#include "NumericalAlgorithms/Spectral/Mesh.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "PointwiseFunctions/Hydro/EquationsOfState/EquationOfState.hpp"
+#include "PointwiseFunctions/Hydro/Tags.hpp"
+#include "Utilities/FakeVirtual.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace grmhd::ValenciaDivClean::subcell {
+/*!
+ * \brief On elements using DG, reconstructs the interface data from a
+ * neighboring element doing subcell.
+ *
+ * The neighbor's packaged data needed by the boundary correction is computed
+ * and returned so that it can be used for solving the Riemann problem on the
+ * interfaces.
+ *
+ * Note that for strict conservation the Riemann solve should be done on the
+ * subcells, with the correction being projected back to the DG interface.
+ * However, in practice such strict conservation doesn't seem to be necessary
+ * and can be explained by that we only need strict conservation at shocks, and
+ * if one element is doing DG, then we aren't at a shock.
+ */
+struct NeighborPackagedData {
+  template <typename DbTagsList>
+  static FixedHashMap<
+      maximum_number_of_neighbors(3), std::pair<Direction<3>, ElementId<3>>,
+      std::vector<double>, boost::hash<std::pair<Direction<3>, ElementId<3>>>>
+  apply(const db::DataBox<DbTagsList>& box,
+        const std::vector<std::pair<Direction<3>, ElementId<3>>>&
+            mortars_to_reconstruct_to) noexcept {
+    using evolved_vars_tag = typename System::variables_tag;
+    using evolved_vars_tags = typename evolved_vars_tag::tags_list;
+    using prim_tags = typename System::primitive_variables_tag::tags_list;
+    using recons_prim_tags = tmpl::push_back<
+        prim_tags,
+        hydro::Tags::LorentzFactorTimesSpatialVelocity<DataVector, 3>>;
+    using fluxes_tags = db::wrap_tags_in<::Tags::Flux, evolved_vars_tags,
+                                         tmpl::size_t<3>, Frame::Inertial>;
+
+    ASSERT(not db::get<domain::Tags::MeshVelocity<3>>(box).has_value(),
+           "Haven't yet added support for moving mesh to DG-subcell. This "
+           "should be easy to generalize, but we will want to consider "
+           "storing the mesh velocity on the faces instead of "
+           "re-slicing/projecting.");
+
+    FixedHashMap<maximum_number_of_neighbors(3),
+                 std::pair<Direction<3>, ElementId<3>>, std::vector<double>,
+                 boost::hash<std::pair<Direction<3>, ElementId<3>>>>
+        nhbr_package_data{};
```

Comment:
[optional] I'd prefer the slightly more explicit/lengthy variable names `reconstructed_prim_tags` instead of `recons_prim_tags` and `neighbor_package_data` instead of `nhbr_package_data`